backend/src/main/resources/db/changelog/db.changelog-master.yaml

```diff
@@ -1209,4 +1212,233 @@ databaseChangeLog:
+                  name: collection_location_name
+                  remarks: The collection location of this specimen type.
+                  type: *string
+              - column:
+                  name: collection_location_code
+                  remarks: The SNOMED code for the collection location of this specimen type.
+                  type: *string
+        - createTable:
+            tableName: device_specimen
+            remarks: A usable combination of device type and specimen type.
+            columns:
+              - column: *pk_column
+              - column: *created_at_column
+              - column: *created_by_column
+              - column: *updated_at_column
+              - column: *updated_by_column
+              - column: *soft_delete_column
+              - column:
+                  name: device_type_id
+                  remarks: The device type for this combination.
+                  type: *idtype
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk__device_specimen__device_type
+                    references: device_type
+              - column:
+                  name: specimen_type_id
+                  remarks: The specimen type for this combination.
+                  type: *idtype
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk__device_specimen__specimen_type
+                    references: specimen_type
+        - addUniqueConstraint:
+            tableName: device_specimen
+            constraintName: uk__device_specimen
+            columnNames: device_type_id, specimen_type_id
+        - createTable:
+            tableName: facility_device_specimen
+            remarks: Many-to-many table for device/specimen configuration at a facility.
+            columns:
+              - column:
+                  name: facility_id
+                  remarks: A facility at which this device/specimen combination is used.
+                  type: *idtype
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk__facility_device_specimen__facility
+                    references: facility
+              - column:
+                  name: device_specimen_id
+                  remarks: A device/specimen combination that is allowed to be used at the facility.
+                  type: *idtype
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk__facility_device_specimen__device_specimen
+                    references: device_specimen
+        - addUniqueConstraint:
+            tableName: facility_device_specimen
+            constraintName: uk__facility_device_specimen
+            columnNames: facility_id, device_specimen_id
+        - addColumn:
+            tableName: facility
+            columns:
+              - column:
+                  name: default_device_specimen_id
```

Comment:
should this be non-nullable?

Reply:
That is a slightly vexed question. If you look at the column it is replacing, you'll find that it is nullable. If you look at the TestOrder business logic, you'll find that it assumes that Facility.defaultDeviceType/defaultDeviceSpecimen has a non-null value. This represents an uneasy peace between two mutually incompatible business requirements: I am inclined not to disturb the equilibrium right now, but you're right to be concerned.

Reply:
uhhhhhh 😬

Reply:
"If you look at the column it is replacing, you'll find that it is nullable." --> are we in a position to populate all the values of the new column with non-null values at this point? if so, we should.

Reply:
Likely but not guaranteed—it's user-populated data, and as noted, the column from which we would be populating it is nullable.

I agree that this is an issue: I do not agree that it is an issue that should make us introduce a possible breakage into an already fairly complicated migration process.